RadarSDK/RadarLogBuffer.h

```diff
@@ -0,0 +1,31 @@
+//
+//  RadarLogBuffer.h
+//  RadarSDK
+//
+//  Copyright © 2021 Radar Labs, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "Radar.h"
+#import "RadarLog.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface RadarLogBuffer : NSObject
+
+- (instancetype)init;
+
++ (instancetype)sharedInstance;
+
+- (void)write:(RadarLogLevel)level message:(NSString *)message;
+
+- (void)purgeOldestLogs;
+
+- (void)clearSyncedLogsFromBuffer:(NSUInteger)numLogs;
+
+- (NSArray<RadarLog *> *)getFlushableLogs;
```

Comment:
Not sure if we are trying to be consistent with Android - but based on feedback from @jkao on my original PR, this was changed to `getFlushableLogsStash` on my implementation.

Reply:
On iOS, it should be a property (`flushableLogs`; no `get`), backed by a computed function.

Reply:
ok tried to update to property style but let me know if had something different in mind

Reply:
Yep--that's great!